pages/transactions/index.tsx

```diff
@@ -9,19 +9,121 @@ import { Button,
    FormControl,
    InputLabel,
    Select,
-   MenuItem}
+   MenuItem,
+   Popover,
+   TextField,
+   RadioGroup,
+   FormControlLabel,
+   Radio,
+   Accordion,
+   AccordionSummary,
+   AccordionDetails,
+   Autocomplete,}
 from "@mui/material";
 import styles from './Transactions.module.css';
-import itemstyles from '../../components/TransactionItem/TransactionItem.module.css';
 import { TabPanel } from '../../components/TabPanel/TabPanel';
 import Icon from '../../assets/Icon';
-import { Transaction } from '../../types/schema';
 import { TransactionList } from '../../components/TransactionList/TransactionList';
-import { getAllTransactions } from '../../firebase/firestore/transaction'; 
-import { getAdmin } from '../../firebase/firestore/admin';
-import { getUser } from '../../firebase/firestore/user';
+import { getAllTransactions, addTransaction } from '../../firebase/firestore/transaction'; 
+import { getAllUsers } from '../../firebase/firestore/user';
+import { Transaction } from '../../types/schema';
  
 const TransactionsPage: React.FunctionComponent = () => {
+    const [addAnchorEl, setAddAnchorEl] = React.useState(null);
+    const [uploadAnchorEl, setUploadAnchorEl] = React.useState(null);
+    const [allUsers, setUsers] = React.useState([]);
+    const [allTransactions, setTransactions] = useState([]);
+    const [success, setSuccess] = useState(false);
+    const [selectedUser, setSelectedUser] = useState('Select User')
+    const [expanded, setExpanded] = useState(false);
+    const [addUser, setAddUser] = useState(null);
+    const [addPoints, setAddPoints] = useState('10');
+    const [addType, setAddType] = useState('');
+    const [addMessage, setAddMessage] = useState('');
+
+    useEffect(() => {
```

Comment:
After thinking about scalability/response time, I think we should find a better way to load data! Even with just the small amount of transactions we have right now, I can notice a slight delay before transactions are properly loaded. I think a better approach would be using NextJS's `GetServerSideProps` or SSR (Server Side Rendering, and here's some documentation to get you started https://nextjs.org/docs/basic-features/data-fetching). This will essentially allow us to load all necessary transactions/data before the pages is loaded/client sees it. If you need any help with this, please let me know! I also did an example of this in my auth PR.

Reply:
uwu!